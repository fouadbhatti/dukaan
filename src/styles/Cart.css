.bag {
  background: #000;
  color: #fff;
  width: 24px;
  height: 27px;
  text-align: center;
  padding-top: 5px;
  margin-top: 8px;
  cursor: pointer;
  position: relative;
  font-size: 12px;
  user-select: none;
}

.bag:before {
  content: "";
  border: 1px solid #000;
  background: #0000;
  width: 14px;
  height: 16px;
  border-radius: 100%;
  position: absolute;
  top: -9px;
  left: 5px;
  z-index: -1;
}

.popover {
  background: transparent;
}

.popover > span {
  pointer-events: none;
}

.overlay {
  width: 100%;
  height: 20px;
  position: absolute;
  top: -20px;
}

.body {
  max-height: 200px;
  overflow: auto;
  background: #f1f1f1;
}

.body > div {
  margin: 0 !important;
  padding: 0 !important;
}

.body > div:not(:last-child):after {
  content: "";
  display: block;
  width: 100px;
  height: 1px;
  background: #e0e0e0;
  margin: 12px;
}

.body > div img {
  width: 64px !important;
}

.body > div h4 {
  font-size: 14px !important;
  white-space: nowrap;
  margin-bottom: 0;
}

.actions {
  padding: 8px 6px;
  background: #f1f1f1;
  border-top: 1px solid #ccc;
}

.actions > span {
  font-weight: bold;
}

.actions > a {
  background: #800080;
  text-decoration: none;
  color: #fff !important;
  padding: 4px 8px;
  border-radius: 4px;
  font-size: 10px;
  position: absolute;
  right: 8px;
  display: inline;
}
