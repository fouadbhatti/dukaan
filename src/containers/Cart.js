import React from 'react';
import { connect } from 'react-redux';
import { Button, Popover, PopoverHeader, PopoverBody } from 'reactstrap';
import Item from '../components/Item';
import { checkout } from '../actions';
import NumberFormat from 'react-number-format';
import styles from '../styles/Cart.css';

class ItemCart extends React.Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      popoverOpen: false
    };

    this.onMouseEnter = this.onMouseEnter.bind(this);
    this.onMouseLeave = this.onMouseLeave.bind(this);

  }

  toggle() {
    this.setState({
      popoverOpen: !this.state.popoverOpen
    });
  }

  onMouseEnter(e) {
    this.setState({
      popoverOpen: true
    });
  }

  onMouseLeave(e) {
    this.setState({
      popoverOpen: false
    });
  }

	render() {
    return (
      <div onMouseEnter={this.onMouseEnter}>
        <div id={'popover-cart'} onClick={this.toggle} onMouseLeave={this.onMouseLeave}>
          <div className={styles.bag}>
            <i>{this.props.cart.count}</i>
          </div>
        </div>
        <Popover className={styles.popover} onMouseEnter={this.onMouseEnter} onMouseLeave={this.onMouseLeave} placement={'bottom'} isOpen={this.state.popoverOpen} target={'popover-cart'} toggle={this.toggle}>
          <div className={styles.overlay}></div>
          <PopoverBody className={styles.body}>
            { this.props.items.map(item => (<Item key={item.id} {...item} children={<ItemActions quantity={item.quantity} />} />))}
            { !this.props.items.length && ' ITEMS' }
          </PopoverBody>

          { this.props.items.length != 0 && <div className={styles.actions}>
            <span>
            TOTAL <NumberFormat
              displayType={'text'}
              value={this.props.cart.price}
              thousandSeparator={true}
              suffix={' AED'} />
            </span>
            <Button size="sm" onClick={() => this.props.dispatch(checkout(this.props.items))}>Secure Checkout</Button>
          </div> }
        </Popover>
      </div>
    );
  }
}

const ItemActions = ({ quantity }) => {
  return (
    <div> Quantity: {quantity}</div>
  )
}

const mapStateToProps = (store) => ({
  cart: store.cart,
  items: store.cart.items
});

const mapDispatchToProps = (dispatch) => ({ dispatch })

const Cart = connect(mapStateToProps, mapDispatchToProps)(ItemCart);

export default Cart;
