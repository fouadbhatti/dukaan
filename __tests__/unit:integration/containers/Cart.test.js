import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import Cart from '../../../src/containers/Cart';

describe('<Cart />', () => {
  let store = {};

  beforeAll(() => {
    const middlewares = [];
    const mockStore = configureStore(middlewares);
    store = mockStore({ cart: { items: [] } });
  });

  it('should render without crashing', () => {
    const wrapper = shallow(<Cart />, { context: { store } });

    expect(wrapper).to.be.a('object');
  });

  it('should have a match snapshot', () => {
    const wrapper = shallow(<Cart />, { context: { store } });

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
