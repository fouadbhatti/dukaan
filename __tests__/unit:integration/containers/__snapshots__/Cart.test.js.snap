// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> should have a match snapshot 1`] = `
<Cart
  cart={
    Object {
      "items": Array [],
    }
  }
  dispatch={[Function]}
  items={Array []}
/>
`;
